### ISSUE TITLE
Deep Linking:  Appointments 

### ISSUE THREAD

**Issue by kellylein** [department-of-veterans-affairs/va-mobile-app#6731]
# Project Status Overview

### Project Objective:

**Goal:** Implement Deep Linking Functionality to **appointments** so that Veterans can more quickly get to their appointment details and save time.

**Deep Linking MVP Reference**: [ Epic for  Secure Messages
](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/6037)

 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
| QA Blocker(Sprint 83)- Need to set up test accounts in Prod and will need coordination with MHV team. Corresponding Thread [HERE](https://dsva.slack.com/archives/C018V2JCWRJ/p1698327707715399) |   |
| ENG Blocker (Sprint 83)- ENG working with Apts ApI team to retrieve data from the Upstream service to merge to our API. Once resolved, Implementation will be unblocked |ENG- EST FIX- 11/3/23  |
| QA Blocker REMOVED (Sprint 85) |Successfully tested in prod with vanotify’s help after a lot of snafu’s. We quickly followed up with some success/failure analytics that should go out with the next release.' (Added to the 2.16.0 release, follow ups to be added in 2.18.0)   |

</details>

### Project Timeline
<details>
<summary>Milestones</summary>

|Step | Timing | 
|-------| ---------| 
|Sprint 0 - Discovery |    |
|Step 1 - Strategy | Q4- Sprint 82  |
|Step 2 - Design  | Q4- Sprint 83  |
|Step 3 - Building / Implementing | Q4-Sprint 83  |
| 100% Release |2.16.0 Release |
|Step 4 - Post Verification|Analytic Follow Ups to be added in 2.18.0 Release (Sprint 85) |
</details>

**Note on Testing:** Testing done on Prod accounts/Results from ENG: " "Successfully tested in prod with vanotify’s help after a lot of snafu’s. We quickly followed up with some success/failure analytics that should go out with the next release.' ( **Implementation originally Added to the 2.16.0 release, follow ups to be added in 2.18.0**)

### Monthly Update 
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|July '23| N/A
|Aug '23| N/A
|Sep '23|N/A
|Oct '23|GREEN- Project Kick off
|Nov '23|BLOCKERS NOTED above
|Nov '23|Sprint 85-Blockers Removed/Testing in Prod Complete (Release in 2.16.0, Analytic Updates will be in 2.18.0)
</details>

# Detailed Project Scope 

### Problem Statement

Today, the Deep Linking functionality does not exist within the appointment space. We have the opportunity to   Implement Deep Linking Functionality to appointments so that Veterans can more quickly get to their appointment details and save time.

### Definition of Done 
<details>
<summary>Following needs to be true</summary>

|Item | Completed | 
|-------| ---------|
| | | 
| | | 

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| | | 
| | | 

</details>

### Important Project Links 
<details>
<summary>Links</summary>

- [Github](url)
- [Product Brief](url)
- [Product Canvas](url)
- Design 
   - [Discovery](url)
   - [Mockups](url)
   - [Final Design](url)
   - [Figma File](url)
- Research
   - [Document 1](url) 
   - [Document 2 ](url) 
-Analytics
   - [Document 1](url) 
</details>

### Team Members

<details>
<summary>Contact List</summary>

- **PM:** Rachel Han
- **Ad Hoc PM:** 
- **Engineering:** 
- **Design:** 
- **External Team 1 & Role:**
- **External Team 1 & Role:**

</details>

### OKR and Analytic Metrics 
<details>
<summary>Metrics</summary>

|What we are measuring | Why | Event Name 
|-------| ---------| ----- |
| | | |
| | | |

</details>

# Checklist

<details>
  <summary>Toggle Product Checklist</summary>

**Project Kickoff & Requirements Gathering**
   - [ ] Project Kickoff session
   - [ ] Initiative Scope finalized
   - [ ] Link Product Brief - [Sprint 0 ](https://docs.google.com/document/d/1WHDJ1iCt9J-acLx8_DSL4tkRKoVYqJGaRUqXuFf7ceg/edit# )
   - [ ] Link to Implementation Strategy
       - Implementation Plan
       - UAT Plan
       - Release Plan
       - Metrics Identified (what metrics are we capturing, how do they tie into stakeholder OKRs, etc)

**Pre-Production Readiness**
- [ ] Associate Readiness
  - [ ] Update product guide
  - [ ] Coordinate with contact center support teams
- [ ] What’s New Content
- [ ] App Store/In App messaging (if applicable)
- [ ] Coordinate with Release coordinator
- [ ] App store Content changes required? Y/N
- [ ] App store images needed? Y/N
- [ ] QA Signed off 
- [ ] UAT complete
- [ ] Obtain Go/No Go decision

**Feature Go Live**
- [ ] Ensure Waygate/Feature toggle release ticket submitted w/engineering & QA 
  - **Template: Turn on Waygate/Turn off feature flag**
- [ ] Coordinate with content & release coordinator if what’s new in app is required
</details>

<details>
  <summary>Toggle Team Collaboration Checklist</summary>

- [ ] Design Intent & Feasibility discussions
- [X] Research needed? No
- [ ] UAT Needed? Yes/No
- [ ] Ticket creation 
- [ ] Demo feature (mid check, end of development)
</details>

<details>
  <summary>Toggle Design/UX Checklist</summary>

- [ ] Review problem/opportunity statement
- [ ] Add questions and assumptions to product brief
- [ ] Add research links to product brief
- [ ] Document high-level research/design plans
- [ ] Create lo/hi fi wireframes & user flows if applicable
- [ ] Add links to Epic & applicable GitHub folders
- [ ] Socialize designs
- [ ] Sign off received
- [ ] Complete Design QA
- [ ] Does it require component review?
- [ ] Accessibility review needed?

</details>

<details>
  <summary>Toggle Content Checklist</summary>

- [ ] Review product brief and surface any questions, assumptions & risks
- [ ] Participate in project discovery and kickoff activities/ceremonies
- [ ] Complete comparative analysis/content research activities for net-new content
- [ ] Review current content and determine where app may need to differ and potential improvements to share with web
- [ ] Review past VA research and decisions documentation
- [ ] Support Research & UX Design in research sessions and synthesis
- [ ] Make content recommendations for lo-/hi-fi wireframes in collaboration with UX Design
- [ ] Collaborate with Sitewide Content team for alignment, improvements, and sign-off
- [ ] Participate in FE hand off
- [ ] Complete content QA
- [ ] Write copy for What's New In App
- [ ] Write copy for app stores' What's New sections.
- [ ] Write copy for updated screenshots in app stores 

</details>

<details>
  <summary>Toggle Engineering Checklist</summary>

- [ ] Are UI designs/ specs available/ready? If yes, they should be attached to  attach specs. (We are assuming the design specs have been approved)
- [ ] Are there any deadlines that need to be met?
- [ ] What is the roll out plan? (Do we need a feature toggle? is this going to be available to every user immediately or do we want to roll out to a few users first)
- [ ] Does this project require api integration? if yes, what api endpoint are we integrating with and what is the expected response body?
- [ ] Do we want to capture analytics for this projects (i.e. update GA, add logs to grafana/datadog etc)?If yes, what metrics would we like to capture?

</details>

<details>
  <summary>Toggle QA Checklist</summary>

- [ ] Create test plan
  -  **Template: QA Test Plan**
- [ ] Identify test data needs
- [ ] Gotten & incorporated test plan review
- [ ] QA Complete
- [ ] Waygate tasks complete
- [ ] Added new feature cases to the RC in TestRail
- [ ] Added feature cases to appropriate section of active cases in TestRail
- [ ] Spun up a (future) UI automation ticket for new feature
</details>

**Comment by StacyB2023** [comment]
@ajsarkar28 Project timeslines updated in the epic.

**Comment by StacyB2023** [comment]
Removed "At risk" and "blocked" tags. Latest testing in prod updates from ENG: ": "Successfully tested in prod with vanotify’s help after a lot of snafu’s. We quickly followed up with some success/failure analytics that should go out with the next release.' (**This was Added to the 2.16.0 release, follow ups to be added in 2.18.0**)"

**Comment by StacyB2023** [comment]
Closing this epic as this was merged with the 2.1.7 release.
Project Updates:

Deep Linking ENG Status 11/28: We successfully tested in prod with vanotify’s help. We quickly followed up with some success/failure analytics that should go out with the next release. 2.17

Analytics: Added this blanket event on 11/28-  T **vama_fail analytic**

Next up: Working with QA's blessing we could probably sneak a new event counter into the release branch though. We want to add a unique one for push notification fails in the 2.19 release

This thread: https://adhoc.slack.com/archives/C02N65QRRB7/p1701961268521139
